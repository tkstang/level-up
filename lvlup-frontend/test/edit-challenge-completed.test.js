import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import EditChallengeApproved from '../src/components/Admin/challenges/edit-challenge/edit-challenge-completed';
import lvlupApp from '../src/reducers/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('EditChallengeApproved should render', () => {
  const store = mockStore({ lvlupApp });
  const edit = shallow(
    <EditChallengeApproved editedChallenge={{ data: { name: 'Feed Hamid', point_value: 5, description: 'Feed your favorite instructor with the evil laugh' } }} store={store} />,
  );
  expect(shallowToJson(edit)).toMatchSnapshot();
});
