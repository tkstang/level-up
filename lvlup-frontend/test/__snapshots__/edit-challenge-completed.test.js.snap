exports[`test EditChallengeApproved should render 1`] = `
<Container
  className="lvl-table">
  <Table
    as="table"
    celled={true}
    color="orange"
    selectable={true}>
    <TableHeader
      as="thead">
      <TableRow
        as="tr"
        cellAs="td">
        <TableHeaderCell
          as="th"
          className="completed-thead"
          textAlign="center">
          Challenge Edit Successful!
        </TableHeaderCell>
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody">
      <TableRow
        as="tr"
        cellAs="td">
        <TableCell
          as="td">
          <Label
            ribbon={true}>
            Challenge
          </Label>
          Feed Hamid
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td">
        <TableCell
          as="td">
          <Label
            ribbon={true}>
            Point Value
          </Label>
          5
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td">
        <TableCell
          as="td">
          <Label
            ribbon={true}>
            Description
          </Label>
          Feed your favorite instructor with the evil laugh
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
  <Grid
    centered={true}>
    <Link
      replace={false}
      to="/admin/challenges">
      <Button
        as="button"
        basic={true}
        color="orange"
        id="completed-btn">
        Back to Challenges
      </Button>
    </Link>
  </Grid>
</Container>
`;
